src/svm/elf.zig

```diff
@@ -549,4 +669,23 @@ pub const Elf = struct {
     pub fn getShdrByName(self: Elf, name: []const u8) ?elf.Elf64_Shdr {
         return self.data.getShdrByName(self.headers, name);
     }
+
+    fn inRangeOfShdr(self: *const Elf, index: usize, addr: usize) bool {
+        const shdr = self.headers.shdrs[index];
+        const sh_offset = shdr.sh_offset;
+        const sh_size = shdr.sh_size;
+        return addr >= sh_offset and addr < sh_offset + sh_size;
+    }
+
+    fn inRangeOfPhdrVm(phdr: elf.Elf64_Phdr, addr: usize) bool {
+        const p_vaddr = phdr.p_vaddr;
+        const p_memsz = phdr.p_memsz;
+        return addr >= p_vaddr and addr < p_vaddr + p_memsz;
+    }
+
+    fn safeSlice(base: anytype, start: usize, len: usize) error{OutOfBounds}!@TypeOf(base) {
+        if (start >= base.len) return error.OutOfBounds;
+        if (start +| len >= base.len) return error.OutOfBounds;
```

Comment:
```suggestion
        if (start +| len > base.len) return error.OutOfBounds;
```

Also does the saturating add actually give us a sensible return value? Why not return error.OutOfBounds if the add would overflow?

Reply:
Nothing, this is a good point as well.

Reply:
Fixed here: 1bedb6f98c70f8767ee6af2bf765af04b8e242ce

Reply:
Did you notice my suggestion to change `>=` to `>`?

In your latest edition you have this:
```zig
        if (end >= base.len) return error.OutOfBounds;
```

`>=` seems wrong to me because it prohibits `base[0..base.len]` which is valid. I think it should be this:

```zig
        if (end > base.len) return error.OutOfBounds;
```